mcqa/data/utils.py

```diff
@@ -1,139 +1,375 @@
 
 
-def _truncate_seq_pair(tokens_a, tokens_b, max_length):
-    """Truncates a sequence pair in place to the maximum length.
+class DataProcessor(object):
+    """Base class for data converters for multiple choice data sets."""
 
-    This is a simple heuristic which will always truncate the longer sequence
-    one token at a time. This makes more sense than truncating an equal percent
-    of tokens from each, since if one sequence is very short then each token
-    that's truncated likely contains more information than a longer sequence.
+    def get_train_examples(self, data_dir):
+        """Gets a collection of `InputExample`s for the train set."""
+        raise NotImplementedError()
 
-    Arguments:
-        tokens_a {[str]} -- First sequence
-        tokens_b {[str]} -- Second sequence
-        max_length {int} -- Maximum length of the sequence pair
-    """
+    def get_dev_examples(self, data_dir):
+        """Gets a collection of `InputExample`s for the dev set."""
+        raise NotImplementedError()
 
-    while True:
-        total_length = len(tokens_a) + len(tokens_b)
-        if total_length <= max_length:
-            break
-        if len(tokens_a) > len(tokens_b):
-            tokens_a.pop()
-        else:
-            tokens_b.pop()
+    def get_test_examples(self, data_dir):
+        """Gets a collection of `InputExample`s for the test set."""
+        raise NotImplementedError()
 
+    def get_labels(self):
+        """Gets the list of labels for this data set."""
+        raise NotImplementedError()
 
-def select_field(features, field):
-    """Select a field from the features
 
-    Arguments:
-        features {InputFeatures} -- List of features : Instances
-        of InputFeatures with attribute choice_features being a list
-        of dicts.
-        field {str} -- Field to consider.
+class RaceProcessor(DataProcessor):
+    """Processor for the RACE data set."""
 
-    Returns:
-        [list] -- List of corresponding features for field.
-    """
+    def get_train_examples(self, data_dir):
+        """See base class."""
+        logger.info("LOOKING AT {} train".format(data_dir))
+        high = os.path.join(data_dir, "train/high")
+        middle = os.path.join(data_dir, "train/middle")
+        high = self._read_txt(high)
+        middle = self._read_txt(middle)
+        return self._create_examples(high + middle, "train")
 
-    return [
-        [
-            choice[field]
-            for choice in feature.choices_features
-        ]
-        for feature in features
-    ]
+    def get_dev_examples(self, data_dir):
+        """See base class."""
+        logger.info("LOOKING AT {} dev".format(data_dir))
+        high = os.path.join(data_dir, "dev/high")
+        middle = os.path.join(data_dir, "dev/middle")
+        high = self._read_txt(high)
+        middle = self._read_txt(middle)
+        return self._create_examples(high + middle, "dev")
 
+    def get_test_examples(self, data_dir):
```

Comment:
Only the name the string "train/dev/test" changes between these functions `get_test_examples, get_dev_examples, get_train_examples`. 

This can be easily made into a either a single method. Or a helper method that is called by the aforementioned three methods.